docs/test-content.md

```diff
@@ -0,0 +1,152 @@
+| `sendMessageCountUnacked` | dict | Arbitrary content to inflate the message size for testing.  See [metrics.md](./metrics.md) for details |
+
+## test reported properties
+
+Reported properties are tested by adding an object to `properties/reported/thief/testContent/reportedPropertyTest` with a pair of `serviceAckId` guids.
+When the service app observes the add of the property, it returns the `addServiceAckId` to the device app inside a `serviceAckResponse` c2d message.
+When the device receives the `addServiceAckId` back from the service, it proceeds to remove the property.
+When the service app observes the removal of hte property, it returns the `removeServiceAckId` to the device app inside a `saerviceAckResponse` c2d message.
+
+```json
+  {
+    "reported": {
+      "thief": {
+        "testContent": {
+          "reportedPropertyTest": {
+            "prop_2": {
+              "addServiceAckId": "1c984433-5593-4a62-bfbf-b4dae6c72bb9",
+              "removeServiceAckId": "18fdb3ec-50fd-4406-a575-8d81a46ed3a1"
+            },
+            "prop_3": {
+              "addServiceAckId": "d0f4dea5-e937-4e59-bd08-3efcfff452e0",
+              "removeServiceAckId": "891cdf2d-992a-4e8d-a7de-ccb624da7796"
+            }
+          }
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `reportedPropertyTest` | dict | object containing all test content for testing reported properties |
+| `prop_x` | dict | object representing a single reported property operation. First operation is held in `prop_1`, followed by `prop_2` and so on. |
+| `addServiceAckId` | guid | `serviceAckId` to return to the device app when the service app first observes this property being added. |
+| `removeServiceAckId` | guid | `serviceAckId` to return to the device app when the service app first observes this property being removed. |
+
+## test c2d messages
+
+When c2d testing is enabled (via `properties/reported/thief/testControl/c2d`), the service sends a continuous sream of c2d messages using the configuration stored in the `testControl/c2d` structure.
+The device app uses the `testC2dMessageIndex` value to keep track of messages received from the service, including dropped messages.
+
+```json
+  {
+    "thief": {
+      "cmd": "testC2d",
+      "serviceRunId": "3e5917f2-3625-4431-92fa-45b184a25498",
+      "pairingId": "64186b84-48d3-4831-8f09-85edc31dc133",
+      "firstMessage": true,
+      "testC2dMessageIndex": 0,
+      "filler": "5EGILSQJHVG7T3WK8N59ITEJZRAS1IGEGCRMC3MGA"
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `cmd` | string | must be `testC2d` |
+| `serviceRunId` | guid |  `runid` for the service app sending this message |
+| `pairingId` | guid | `pairingid` representing the device/service app pairing |
+| `firstMessage` | boolean | `true` if this is the first message in the sequence.  `false` otherwise. |
```

Comment:
grammatically: `in the sequence, otherwise `false`. (comma not period and incomplete sentence)